src/ds/champmap.h

```diff
@@ -189,6 +203,23 @@ namespace champ
       return node_as<SubNodes<K, V, H>>(c_idx)->get(depth + 1, hash, k);
     }
 
+    const V* getp(SmallIndex depth, Hash hash, const K& k) const
+    {
+      const auto idx = mask(hash, depth);
+      const auto c_idx = compressed_idx(idx);
+
+      if (c_idx == (SmallIndex)-1)
+        return nullptr;
+
+      if (data_map.check(idx))
+        return node_as<Entry<K, V>>(c_idx)->getp(k);
+
+      if (depth == (collision_depth - 1))
+        return node_as<Collisions<K, V, H>>(c_idx)->getp(hash, k);
+
+      return node_as<SubNodes<K, V, H>>(c_idx)->getp(depth + 1, hash, k);
+    }
```

Comment:
To avoid duplicating this logic, can we update the implementation of `std::optional<V> get` to just call `getp`? Will make it clear that the lookup is identical, and the only difference is the extra copy in the `optional` version.

Reply:
Definitely. Just pushed the dead code removal.